torch/_meta_registrations.py

```diff
@@ -3643,8 +3645,52 @@ def meta_relu_(self):
     return self
 
 
+def check_index_put_inputs(self, indices, values, accumulate=False):
```

Comment:
How does one verify this matches the non-meta behavior?  Is this based on some checking in the actual index_put kernel?

Reply:
Thats correct. The kernels for index.Tensor and index_put both use the same logic to verify the inputs and this logic has been replicated in Python for index.Tensor, see meta_index_Tensor in this file. check_index_put_inputs just adds some initial checking due to the difference between index_put and index.Tensor before calling meta_index_Tensor.

Reply:
I am asking for some help here as a reviewer.  Reading this code I don't see how I'm supposed to verify this matches the C++ implementation.

Reply:
Both index.Tensor and index_put use the same logic (using the function make_info) when verifying the indicies and values.

The meta kernel for index.Tensor is [here](https://github.com/pytorch/pytorch/blob/93372455a73043332c16a71cb9dccdf3e0412a57/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L430) and the kernel calls the function [make_info](https://github.com/pytorch/pytorch/blob/93372455a73043332c16a71cb9dccdf3e0412a57/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L458C1-L459C1) to do most of the input verification.

Verification for the index_put inputs is also done by calling make_info. The index_put kernel can be found [here](https://github.com/pytorch/pytorch/blob/93372455a73043332c16a71cb9dccdf3e0412a57/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L708) and the call to make_info [further down](https://github.com/pytorch/pytorch/blob/93372455a73043332c16a71cb9dccdf3e0412a57/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L740). 

The make_info function is implemented [here](https://github.com/pytorch/pytorch/blob/93372455a73043332c16a71cb9dccdf3e0412a57/aten/src/ATen/native/TensorAdvancedIndexingUtils.h#L57)

The make_info function has been replicated in python: https://github.com/pytorch/pytorch/blob/93372455a73043332c16a71cb9dccdf3e0412a57/torch/_meta_registrations.py#L2786

Previously the python meta kernel for index_put did not verify that the indices and values are valid (e.g. running the checks in make_info). This PR enables that by calling into the replicated python logic for make_info